share/completions/gcc.fish

```diff
@@ -482,7 +482,11 @@ complete -c gcc -s c -d 'Compile or assemble the source files, but do not link.'
 complete -c gcc -s S -d 'Stop after the stage of compilation proper; do not assemble'
 complete -c gcc -s E -d 'Stop after the preprocessing stage; do not run the compiler proper'
 complete -c gcc -o llibrary -d 'Search the library named library when linking'
+for i in (ls /usr/lib/ | string match -r -g "^lib(.*?)\.so.*")
```

Comment:
Calling ls here is awkward for a variety of reasons - it's an external command that may or may not escape its output or be aliased to any number of other things.

Use

```fish
for i in (path basename /usr/lib/lib*.so* | string match -r -g "^lib(.*?)\.so.*")
```

Which is also easily extendable to other directories - I would assume that /lib and the "lib64" variants are also of interest